datajunction-clients/python/README.md

```diff
@@ -16,130 +24,346 @@ from datajunction import DJClient
+```
+
+**NOTE:**
+The `cascade` parameter in both of above methods allows for cascading
+effect applied to all underlying nodes and namespaces. Use it with caution!
 
 ### Nodes
 
+To learn what **Node** means in the context of DJ, please check out [this datajuntion.io page](https://datajunction.io/docs/0.1.0/dj-concepts/nodes/).
+
+To list all (or some) nodes in the system you can use the `list_<node-type>()` methods described
+in the **DJ Client : Basic Access** section or you can use the namespace based method:
+
 All nodes for a given namespace can be found with:
 ```python
-dj.namespace("default").nodes()
+djb.namespace("default").nodes()
 ```
 
 Specific node types can be retrieved with:
 ```python
-dj.namespace("default").sources()
-dj.namespace("default").dimensions()
-dj.namespace("default").metrics()
-dj.namespace("default").transforms()
-dj.namespace("default").cubes()
+djb.namespace("default").sources()
+djb.namespace("default").dimensions()
+djb.namespace("default").metrics()
+djb.namespace("default").transforms()
+djb.namespace("default").cubes()
 ```
 
 To create a source node:
 
 ```python
-from datajunction import NodeMode
-
-repair_orders = dj.new_source(
+repair_orders = djb.create_source(
     name="repair_orders",
     display_name="Repair Orders",
     description="Repair orders",
     catalog="dj",
     schema_="roads",
     table="repair_orders",
 )
-repair_orders.save(mode=NodeMode.PUBLISHED)
 ```
 
-Nodes can also be created as drafts with:
+Nodes can also be created in draft mode:
+
 ```python
+from datajunction import NodeMode
+
 repair_orders.save(mode=NodeMode.DRAFT)
 ```
 
 To create a dimension node:
+
 ```python
-repair_order = dj.new_dimension(
-    name="repair_order",
+repair_order = djb.create_dimension(
+    name="default.repair_order_dim",
     query="""
     SELECT
       repair_order_id,
       municipality_id,
       hard_hat_id,
       dispatcher_id
-    FROM repair_orders
+    FROM default.repair_orders
     """,
     description="Repair order dimension",
     primary_key=["repair_order_id"],
 )
-repair_order.save()
 ```
 
 To create a transform node:
 ```python
-large_revenue_payments_only = dj.new_transform(
-    name="large_revenue_payments_only",
+large_revenue_payments_only = djb.create_transform(
+    name="default.large_revenue_payments_only",
     query="""
     SELECT
       payment_id,
       payment_amount,
       customer_id,
       account_type
-    FROM revenue
+    FROM default.revenue
     WHERE payment_amount > 1000000
     """,
     description="Only large revenue payments",
 )
-large_revenue_payments_only.save()
 ```
 
 To create a metric:
 ```python
-num_repair_orders = dj.new_metric(
-    name="num_repair_orders",
+num_repair_orders = djb.create_metric(
+    name="default.num_repair_orders",
     query="""
     SELECT
       count(repair_order_id)
     FROM repair_orders
     """,
     description="Number of repair orders",
 )
-num_repair_orders.save()
 ```
+
+### Reference
```

Comment:
Maybe we can eventually use something like pydoc to automatically generate this kind of reference documentation for the entire client library.

Reply:
That would be fantastic.